pkg/webui/console/containers/gateways-table/index.js

```diff
@@ -61,23 +68,44 @@ const headers = [
 ]
 
 @bind
-export default class GatewaysTable extends React.Component {
+class GatewaysTable extends React.Component {
 
-  baseDataSelector ({ gateways }) {
-    return gateways
+  baseDataSelector (state) {
+    return {
+      gateways: selectGateways(state),
+      totalCount: selectGatewaysTotalCount(state),
+      fetching: selectGatewaysFetching(state),
+    }
   }
 
   render () {
+    const { error, ...rest } = this.props
+
+    if (error) {
+      throw error
+    }
+
     return (
       <FetchTable
         entity="gateways"
         addMessage={m.add}
         headers={headers}
-        getItemsAction={getGatewaysList}
-        searchItemsAction={searchGatewaysList}
+        getItemsAction={getGateways}
         baseDataSelector={this.baseDataSelector}
-        {...this.props}
+        {...rest}
       />
     )
   }
 }
+
+GatewaysTable.propTypes = {
+  error: PropTypes.error,
+}
+
+GatewaysTable.defaultProps = {
+  error: null,
+}
+
+export default connect(state => ({
```

Comment:
Why is this not a decorator?

Reply:
to avoid proptypes warning because of props passed from `connect`

Reply:
I see. Seems a bit hacky, could we do anything about this?